assets/js/base/context/providers/cart-checkout/checkout-state/actions.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Internal dependencies
+ */
+import { ACTION } from './constants';
+import type { PaymentResultDataType } from './types';
+
+export interface ActionType {
+	type: ACTION;
+	data?:
+		| Record< string, unknown >
+		| Record< string, never >
+		| PaymentResultDataType;
+	url?: string;
+	customerId?: number;
+	orderId?: number;
+	shouldCreateAccount?: boolean;
+	hasError?: boolean;
+	orderNotes?: string;
+}
+
+/**
+ * All the actions that can be dispatched for the checkout.
+ */
+export const actions = {
+	setPristine: (): ActionType => ( {
+		type: ACTION.SET_PRISTINE,
+	} ),
```

Comment:
If we do something like the following, we shouldn't have to define the `ActionType` interface and the types will be more narrowly defined to the actual action?

```suggestion
	setPristine: () => ( {
		type: ACTION.SET_PRISTINE,
	} as const ),
```

I make this suggestion because this was what was recommended to me for action objects when I was doing the TS conversion for the cart data store ([example](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/a5add04cd927090c356136eaf2920b4d6f1827cc/assets/js/data/cart/actions.ts#L31-L39)).

Reply:
I'm not sure if this always works—one of the `as const` examples failed eslint checks when the function contained more logic (see https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4206/files#diff-a942a5198733d5fa79775b811f7cfe45bbf4d144bdf209a3334694286a6e49e4).

Why is const better than typing the props? @opr any thoughts here?

Reply:
For reference, [this is where it was suggested](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/3768#pullrequestreview-581575534).

Reply:
"const assertions" 

I changed this in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4200/commits/c1b972e8ba3ba1da03270a20b4549f265f909efa